build error on ts monorepo

https://github.com/himself65/bunchee-issue/tree/issue-1

```json
{
  "name": "core",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.js",
      "import": "./dist/index.js"
    }
  },
  "scripts": {
    "build": "bunchee"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "bunchee": "^4.3.3"
  }
}

```

```shell
cd packages/core
pnpm run build
```

```
/Users/himself65/Library/Caches/fnm_multishells/9583_1705091499933/bin/npm run build

> build
> bunchee

error TS6379: Composite projects may not disable incremental compilation.

⨯ RollupError: Failed to compile. Check the logs above.
    at Object.error (/Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup@4.9.5/node_modules/rollup/dist/shared/parseAst.js:279:30)
    at Object.error (/Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup@4.9.5/node_modules/rollup/dist/shared/rollup.js:820:32)
    at Object.error (/Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup@4.9.5/node_modules/rollup/dist/shared/rollup.js:18666:42)
    at generateDtsFromTs (/Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup-plugin-dts@6.1.0_rollup@4.9.5_typescript@5.3.3/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.cjs:1857:30)
    at Object.transform (/Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup-plugin-dts@6.1.0_rollup@4.9.5_typescript@5.3.3/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.cjs:1866:38)
    at /Users/himself65/Code/bunchee-monorepo/node_modules/.pnpm/rollup@4.9.5/node_modules/rollup/dist/shared/rollup.js:1005:40

Process finished with exit code 1

```

--- a/src/build-config.ts
+++ b/src/build-config.ts
@@ -183,13 +183,6 @@ async function buildInputConfig(
         declarationMap: false,
         skipLibCheck: true,
         target: 'ESNext',
-        // Some react types required this to be false by default.
-        // Some type package like express might need this as it has other dependencies.
-        // Let users able to toggle this in tsconfig.
-        preserveSymlinks:
-          'preserveSymlinks' in tsCompilerOptions
-            ? tsCompilerOptions.preserveSymlinks
-            : false,
         ...(!tsCompilerOptions.jsx
           ? {
               jsx: 'react-jsx',
@@ -197,7 +190,11 @@ async function buildInputConfig(
           : undefined),
         // error TS5074: Option '--incremental' can only be specified using tsconfig, emitting to single
         // file or when option '--tsBuildInfoFile' is specified.
-        incremental: false,
+        ...(tsCompilerOptions.incremental && !tsCompilerOptions.tsBuildInfoFile
+          ? {
+            incremental: false,
+          } : undefined
+        )
       })
 
     const dtsPlugin = (

--- a/test/integration.test.ts
+++ b/test/integration.test.ts
@@ -18,6 +18,7 @@ const getPath = (filepath: string) => join(integrationTestDir, filepath)
 
 const testCases: {
   name: string
+  dir?: string
   args?: string[]
   before?(dir: string): Promise<void> | void
   expected(
@@ -245,9 +246,7 @@ const testCases: {
         join(dir, './dist/index.js'),
         join(dir, './dist/index.d.ts'),
       ]
-      for (const f of distFiles) {
-        expect(await existsFile(f)).toBe(true)
-      }
+      await assertContainFiles(dir, distFiles)
       expect(await fsp.readFile(distFiles[1], 'utf-8')).toContain(
         'declare function _default(): string;',
       )
@@ -260,9 +259,21 @@ const testCases: {
       // TODO: disable incremental and avoid erroring
       const distFiles = ['./dist/index.js', './dist/index.d.ts']
 
-      for (const f of distFiles) {
-        expect(await existsFile(join(dir, f))).toBe(true)
-      }
+      await assertContainFiles(dir, distFiles)
+      expect(await fsp.readFile(join(dir, distFiles[1]), 'utf-8')).toContain(
+        'declare const _default: () => string;',
+      )
+      expect(await existsFile(join(dir, './dist/.tsbuildinfo'))).toBe(false)
+    },
+  },
+  {
+    name: 'ts-incremental-with-buildinfofile',
+    args: [],
+    async expected(dir) {
+      // TODO: disable incremental and avoid erroring
+      const distFiles = ['./dist/index.js', './dist/index.d.ts']
+      await assertContainFiles(dir, distFiles)
+
       expect(await fsp.readFile(join(dir, distFiles[1]), 'utf-8')).toContain(
         'declare const _default: () => string;',
       )
@@ -276,9 +287,7 @@ const testCases: {
       // should still emit declaration files
       const distFiles = ['./dist/index.js', './dist/index.d.ts']
 
-      for (const f of distFiles) {
-        expect(await existsFile(join(dir, f))).toBe(true)
-      }
+      await assertContainFiles(dir, distFiles)
       expect(await fsp.readFile(join(dir, distFiles[1]), 'utf-8')).toContain(
         'declare const _default: () => string;',
       )
@@ -366,9 +375,7 @@ const testCases: {
         join(dir, './dist/bin.js'),
         join(dir, './dist/bin.d.ts'),
       ]
-      for (const f of distFiles) {
-        expect(await existsFile(f)).toBe(true)
-      }
+      await assertContainFiles(dir, distFiles)
       expect(await fsp.readFile(distFiles[0], 'utf-8')).toContain(
         '#!/usr/bin/env node',
       )
@@ -787,7 +794,7 @@ async function runBundle(
 function runTests() {
   for (const testCase of testCases) {
     const { name, args = [], expected, before } = testCase
-    const dir = getPath(name)
+    const dir = getPath(testCase.dir ?? name)
     test(`integration ${name}`, async () => {
       debug.log(`Command: bunchee ${args.join(' ')}`)
       execSync(`rm -rf ${join(dir, 'dist')}`)

--- a/test/integration/ts-incremental-with-buildinfofile/package.json
+++ b/test/integration/ts-incremental-with-buildinfofile/package.json
@@ -1,0 +1,5 @@
+{
+  "name": "ts-incremental",
+  "types": "./dist/index.d.ts",
+  "exports": "./dist/index.js"
+}

--- a/test/integration/ts-incremental-with-buildinfofile/src/index.ts
+++ b/test/integration/ts-incremental-with-buildinfofile/src/index.ts
@@ -1,0 +1,1 @@
+export default () => 'index'

--- a/test/integration/ts-incremental-with-buildinfofile/tsconfig.json
+++ b/test/integration/ts-incremental-with-buildinfofile/tsconfig.json
@@ -1,0 +1,6 @@
+{
+  "compilerOptions": {
+    "incremental": true,
+    "tsBuildInfoFile": "tsconfig.tsbuildinfo"
+  }
+}

--- a/test/testing-utils.ts
+++ b/test/testing-utils.ts
@@ -24,7 +24,7 @@ export async function existsFile(filePath: string) {
 export function assertContainFiles(dir: string, filePaths: string[]) {
   const results = []
   for (const filePath of filePaths) {
-    const fullPath = path.join(dir, filePath)
+    const fullPath = path.resolve(dir, filePath)
     const existed = fs.existsSync(fullPath)
     if (existed) {
       results.push(filePath)

